weldx/time.py

```diff
@@ -245,8 +245,8 @@ def __init__(
         if isinstance(time, pd.Index) and not time.is_monotonic_increasing:
             raise ValueError("The time values passed are not monotonic increasing.")
 
-        self._time = time
-        self._time_ref = time_ref
+        self._time: Union[pd.TimedeltaIndex, pd.DatetimeIndex] = time
+        self._time_ref: pd.Timestamp = time_ref
```

Comment:
I had no idea we could use type hints just like that in code 👀

Reply:
Me neither, but I find it to be really useful.